ballerina/modules/parser/lexer.bal

```diff
@@ -103,22 +103,124 @@ public class Lexer {
         _ = self.readNextChar(); // Ignore first double quote character
-        while (true) {
+        string value = self.charReader.peek();
+        if value == DOUBLE_QUOTE {
+            value = self.readNextChar();
+            value = self.charReader.peek();
+            if value == DOUBLE_QUOTE {
+                word = self.readBlockStringLiteral();
+                return getToken(word.trim(), T_STRING, location);
+            } else {
+                return getToken(word, T_STRING, location);
+            }
+        } else {
+            while true {
+                value = self.charReader.peek();
+                if value is LineTerminator {
+                    string message = "Syntax Error: Unterminated string.";
+                    return error UnterminatedStringError(message, line = self.currentLocation.line,
+                                                         column = self.currentLocation.column);
+                } else {
+                    value = self.readNextChar();
+                    if value == DOUBLE_QUOTE && previousChar != BACK_SLASH {
+                        break;
+                    } else {
+                        word += value;
+                    }
+                    previousChar = value;
+                }
+            }
+            return getToken(word, T_STRING, location);
+        }
+    }
+
+    isolated function readBlockStringLiteral() returns string {
+        string previousChar = "";
+        string word = "";
+        string[] lines = [];
+        _ = self.readNextChar(); // Ignore third double quote character
+        while true {
             string value = self.charReader.peek();
-            if (value is LineTerminator) {
-                string message = "Syntax Error: Unterminated string.";
-                return error UnterminatedStringError(message, line = self.currentLocation.line,
-                                                     column = self.currentLocation.column);
-            } else if (value == DOUBLE_QUOTE && previousChar != BACK_SLASH) {
+            if value == NEW_LINE {
+                lines.push(word);
+                word = "";
                 value = self.readNextChar();
-                break;
+            } else if value == DOUBLE_QUOTE {
+                string wordWithQuotes = self.readQuotesInBlockStrings(word);
+                if wordWithQuotes == word {
+                    lines.push(word);
+                    break;
+                } else {
+                    word = wordWithQuotes;
+                }
             } else {
                 value = self.readNextChar();
                 word += value;
             }
             previousChar = value;
         }
-        return getToken(word, T_STRING, location);
+        return self.getBlockStringValue(lines);
+    }
+
+    isolated function readQuotesInBlockStrings(string word) returns string {
+        string previousChar = self.readNextChar();
```

Comment:
This variable is redundant. It has only been used to concatenate the character inside the loop. The character is not carried forward for the next iteration. We can use a local variable for that.